proxy/envoy/resources.go

```diff
@@ -416,7 +424,15 @@ type Listener struct {
 type SSLContext struct {
 	CertChainFile  string `json:"cert_chain_file"`
 	PrivateKeyFile string `json:"private_key_file"`
-	CACertFile     string `json:"ca_cert_file,omitempty"`
+	CaCertFile     string `json:"ca_cert_file,omitempty"`
+}
+
+// SSLContextWithSAN definition, VerifySubjectAltName cannot be nil.
+type SSLContextWithSAN struct {
+	CertChainFile        string   `json:"cert_chain_file"`
+	PrivateKeyFile       string   `json:"private_key_file"`
+	CaCertFile           string   `json:"ca_cert_file,omitempty"`
+	VerifySubjectAltName []string `json:"verify_subject_alt_name"`
```

Comment:
Use omitempty and combine with SSLContext. The two structs are almost identical.